lib/bin/lwm2m_carrier/scripts/lwm2m_carrier_divided_dfu.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+import argparse
+import os
+import struct
+import errno
+
+HEADER_LENGTH = 42
+HEADER_MAGIC = 0x424ad2dc
+
+class chunk_header:
+    def __init__(self, number, is_last, offset, version_str):
+        self.__number = number
+        self.__is_last = is_last
+        self.__offset = offset
+        self.__version_str = version_str.encode('utf-8')
+        self.__pack_format = '<IB?I32s'
+
+    def encode(self):
+        """
+        Encode the header, which consists of:
+			- Header magic (4 bytes).
+            - File number (1 byte).
+            - Flag indicating whether the file is the last in the sequence (1 byte).
+            - Offset of the application update data relative to the full sequence (4 bytes).
+            - Application version string (32 bytes).
+        """
+        return struct.pack(self.__pack_format, HEADER_MAGIC, self.__number, self.__is_last, self.__offset, self.__version_str)
+
+def split(image, max_chunk_size):
+    """
+    Split the provided image into chunks of size of up to max_chunk_size bytes.
+    """
+    chunks = []
+
+    with open(image, 'rb') as file:
+        while True:
+            chunk = file.read(max_chunk_size)
+            if not chunk:
+                break
+            chunks.append(chunk)
+
+    return chunks
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Split the application update image into proprietary LwM2M Carrier divided DFU files',
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        allow_abbrev=False)
+
+    parser.add_argument('-i', '--input-file', required=True, help='The application update file to be divided.')
+    parser.add_argument('-v', '--version-str', required=True, help='The application version string to be included in the header (maximum 32 bytes including the null termination).')
+    parser.add_argument('-s', '--max-file-size', default=100, type=int, help='(Optional) The maximum size of a divided DFU file in KB.')
+    parser.add_argument('-o', '--output-dir', required=True, help='The directory to store the divided DFU files into.')
+
+    return parser.parse_args()
+
+def main():
+    # Parse arguments.
+    args = parse_args()
+
+    # Validate the length of the version string.
+    if len(args.version_str) > 31:
+        raise OSError(errno.EINVAL, "Input version string is too long")
```

Comment:
This used to be a problem if the test object has a longer name.
If there is a limitation here, there should be a limitation of the version string inside the lib as well

Reply:
There is validation for null-termination in the library and the header is fixed, so it should be ok.